digitalidentity/service_test.go

```diff
@@ -113,3 +128,42 @@ func createShareUrlWithErrorResponse(statusCode int, responseBody string) (share
 
 	return CreateShareSession(client, &scenario, "sdkId", "https://apiurl", key)
 }
+
+type MockHttpClient struct{}
```

Comment:
we don't really need this.
We can just use the same mechanism as above where teh mockClient has a do function that returns the desired payload
eg
do: func(*http.Request) (*http.Response, error) {
			return &http.Response{
				StatusCode: 201,
				Body:       io.NopCloser(strings.NewReader(`{"status":"success","ref_id":"0"}`)),
			}, nil
		},